### PR TITLE
Add documentation for weapon object natives

### PR DESCRIPTION
**Pull request by jaymo1011**
I got annoyed with how poor and inconstant the documentation was for these natives so I fixed it :P

### CODE REVIEW

**Comment by jaymo1011** [comment]
inconsistent*

I got to actually read before I post things 😛

File: WEAPON/CreateWeaponObject.md [hunk 1]
```diff
@@ -5,21 +5,30 @@ ns: WEAPON
 
 ```c
 // 0x9541D3CF0D398F36 0x62F5987F
-Object CREATE_WEAPON_OBJECT(Hash weaponHash, int ammoCount, float x, float y, float z, BOOL showWorldModel, float heading, Any p7);
```
Comment (blattersturm) [inline WEAPON/CreateWeaponObject.md]: Don't **ever** add parameters to a native definition.

File: WEAPON/RemoveWeaponComponentFromWeaponObject.md [hunk 2]
```diff
@@ -5,11 +5,12 @@ ns: WEAPON
 
 ```c
 // 0xF7D82B0D66777611 0xA6E7ED3C
-void REMOVE_WEAPON_COMPONENT_FROM_WEAPON_OBJECT(Any p0, Any p1);
+void REMOVE_WEAPON_COMPONENT_FROM_WEAPON_OBJECT(Object weaponObject, Hash componentHash);
```
Comment (stannum-cfx) [inline WEAPON/RemoveWeaponComponentFromWeaponObject.md]: Changing from `Any` to `Hash` will break C# scripts that don't use a `uint` there. Please change it to `int` instead to keep compatibility with C# scripts.

Comment (sgottelk) [reply WEAPON/RemoveWeaponComponentFromWeaponObject.md:8]:
```suggestion
void REMOVE_WEAPON_COMPONENT_FROM_WEAPON_OBJECT(Object weaponObject, int componentHash);
```

**Comment by jaymo1011** [comment]
the repository where these changes *were* went bye bye so I'll have to PR again at a later date.

**Pull request closed without merging**